src/zope/server/serverbase.py

```diff
@@ -42,15 +42,19 @@ def __init__(self, ip, port, task_dispatcher=None, adj=None, start=1,
         self.port = port
         self.task_dispatcher = task_dispatcher
         self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
-        self.set_reuse_addr()
-        self.bind((ip, port))
-        self.verbose = verbose
-        self.hit_log = hit_log
-        self.logger = logging.getLogger(self.__class__.__name__)
-        self.server_name = self.computeServerName(ip)
-
-        if start:
-            self.accept_connections()
+        try:
+            self.set_reuse_addr()
+            self.bind((ip, port))
+            self.verbose = verbose
+            self.hit_log = hit_log
+            self.logger = logging.getLogger(self.__class__.__name__)
+            self.server_name = self.computeServerName(ip)
+
+            if start:
+                self.accept_connections()
+        except BaseException:
```

Comment:
I prefer just plain `except:` in the "catch just to cleanup and reraise" scenario, but that's just me.

Reply:
Tools like flake8 complain about bare `except:` clauses so I got into the habit of doing this.

Reply:
Under pylint, `except:` gets "warning: bare-except" but it's smart enough to notice that there's a `raise` in the body and not issue the warning.

Interestingly, I just noticed that while it does issue 'warning: broad-except" for `except Exception:` (with no `raise`) it *doesn't* issue that for `except BaseException:` (with no raise). I always assumed that it would issue 'broad-except'. TIL (thanks). I can't decide if that seems like a bug or not.